src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateScriptsRule.cs

```diff
@@ -113,27 +116,32 @@ internal string FormatScriptCommand(string scriptCommandline, string scriptExten
             return command;
         }
 
-        internal string ReplaceScriptVariables(string command)
+        internal string ReplaceScriptVariables(string scriptCommandline)
         {
-            foreach (var scriptVariableEntry in ScriptVariableToMSBuildMap)
+            Func<string, string> scriptVariableReplacementDelegate = key =>
             {
-                var scriptVariableName = scriptVariableEntry.Key;
-                var msbuildMapping = scriptVariableEntry.Value;
-
-                if (command.Contains($"%{scriptVariableName}%"))
+                if (ScriptVariableToMSBuildMap.ContainsKey(key))
```

Comment:
So, this replacement function will replace if there is a key in the map with a value. If they key is in the map without a value, it will fail? In which case will this happen.

And, if the key is not in the map, it will just return the key?

Should it be called ReplaceScriptVariableIfPossibleDelegate?